rfcs/45-stageless.md

```diff
@@ -0,0 +1,996 @@
+        // new closures that can be used as conditions.
+        .add_system(gravity.run_if(resource_exists(Gravity)))
+        // The systems in this set won't run if `Paused` is `true`.
+        .configure_set(GameSystems::Physics.run_if(resource_equals(Paused(false))))
+        .run();
+}
+```
+
+### Exclusive systems
+
+As much as everyone loves to see systems running in parallel, sometimes a little `&mut World` is necessary.
+Previous versions of Bevy treated these "exclusive" systems as a separate concept, with their own set of types and traits.
+They couldn't have `Local` params and could only be inserted at specific points of a stage.
+
+That is no longer the case.
+Exclusive systems are now just regular systems.
+They can have `Local` params and be scheduled wherever you want.
+
+Command application is now signaled through an exclusive system called `apply_system_buffers`.
+You can add instances of this system anywhere in your schedule.
+If one system depends on the effects of commands from another, make sure an `apply_system_buffers` appears somewhere between them.
+
+```rust
+use bevy::prelude::*;
+
+#[derive(SystemLabel)]
+enum MySystems {
+    /* ... */
+    FlushProjectiles,
+    /* ... */
+}
+
+struct ProjectilePlugin;
+
+impl Plugin for ProjectilePlugin {
+    fn build(app: &mut App) {
+        app
+        .add_system(spawn_projectiles.before(CoreSystems::FlushPostUpdate))
+        .add_systems(
+            chain![
+                check_if_projectiles_hit,
+                despawn_projectiles_that_hit,
+                // Wherever you want commands to be applied, insert an instance of `apply_system_buffers`
+                // Note: You cannot use type-derived labels for ordering if duplicates exist,
+                // so give this system a name if you want to order relative to it elsewhere.
+                apply_system_buffers.named(MySystems::FlushProjectiles)
+                fork_projectiles,
+            ]
+            .after(CoreSystems::FlushPostUpdate)
+            .before(CoreSystems::FlushLast)
+        )
+    }
+}
+```
+
+### Running other schedules on-demand with exclusive systems
+
+Exclusive systems can be used to run other schedules in an on-demand fashion.
+Each system is stored within a `Schedule`, which themselves live within a public `Schedules` resource.
+
+```rust
+fn fancy_exclusive_system(world: &mut World) {
+    // removes the selected schedule from Schedules, executes a fn with it, then returns it to the world
+    world.schedule_scope(ScheduleLabel, |world, mut schedule| {
+        // This runs the entire schedule on the world
+        // Alternatively: world.run_schedule(ScheduleLabel)
+        schedule.run(world);
+        
+        // This runs all systems with the provided label, obeying any internal ordering
+        // Alternatively: world.run_system_from_schedule(SystemLabel, ScheduleLabel)
+        schedule.run_systems(SystemLabel, world);

```

Comment:
While I don't think this will suffer from the "build a schedule on the spot" computational complexity / set math footguns, it seems like this still suffers from the "violating constraints" problem (such as running consumer systems which were configured to run after producer systems), as [outlined in this thread](https://github.com/bevyengine/rfcs/pull/45#discussion_r956489590). I thought we had reached consensus on schedules being the unit of execution, but this is going back to "execution of arbitrary user-defined schedule subsets, which could violate the intent / invariants defined by the schedule authors".